packages/code-du-travail-data/search/conventions_collectives_index.js

```diff
@@ -0,0 +1,145 @@
+const { Client } = require("@elastic/elasticsearch");
+const conventions = require("@socialgouv/kali-data/data/index.json");
+
+const INDEX_NAME = "conventions_collectives";
+
+const client = new Client({
+  node: process.env.ELASTICSEARCH_URL
+});
+
+const deleteIndex = () =>
+  client.indices
+    .delete({
+      index: INDEX_NAME
+    })
+    .then(() => {
+      console.log(`Index ${INDEX_NAME} dropped.`);
+    })
+    .catch(e => {
+      console.error(
+        `Something wrong happenned while deleting the index ${INDEX_NAME}:`,
+        e
+      );
+    });
+
+const createIndex = () =>
+  client.indices
+    .create({
+      index: INDEX_NAME,
+      body: {
+        settings: {
+          number_of_shards: 1,
+          number_of_replicas: 0
+          // index: {
+          //   analysis: {
+          //   }
+          // }
+        },
+        mappings: {
+          properties: {
+            conventionId: {
+              type: "text"
+            }
+          }
+        }
+      }
+    })
+    .then(() => {
+      console.log(`Index ${INDEX_NAME} created.`);
+    })
+    .catch(e => {
+      console.error(
+        `Something wrong happenned while creating the index ${INDEX_NAME}:`,
+        e
+      );
+    });
+
+const batchIndexing = body =>
+  client
+    .bulk({
+      index: INDEX_NAME,
+      body
+    })
+    .then(() => {
+      console.log(
+        `Indexing ${body.length / 2} documents in ${INDEX_NAME} index`
+      );
+    })
+    .catch(e => {
+      console.error(
+        `Something wrong happenned while indexing documents in ${INDEX_NAME}:`,
+        e.body.error
+      );
+    });
+
+(async () => {
+  await deleteIndex();
+  await createIndex();
+
+  let conventionIndex = 0;
+  const batchApproximateSize = 250;
+  // we don't want to have too many documents in memory here
+  // (the json itself weights more than 200MB) so we break
+  // the operation in several smaller batch requests
+  while (conventionIndex < conventions.length) {
```

Comment:
I think you should separate the domain stuff (kali parsing preparing) from the es stuff wich is more generic and could be reuse later (in api for example)
Also there is a batching function that may fit you needs in `dataset/fiches_ministere_travail/utils.js` 
```
async function batchPromise(list, size, callback)) 
```
where callback is a function that return a promise
or batch that will return an array of array 
```
batch(list, size)
```
batch([1,2,3,4,5], 2) => [[1, 2], [3, 4], [5]] 
```

Reply:
I'm 100% open in any suggestions concerning this file ! It is quite experimental but I think Julien wants to keep it to the bare minimal (i.e. in one file) for now.
I'll take a look at the batch function but I'm not sure it's going to fit because `batch(list, size)` imply to have the list in memory already which exceed the memory limits even with some node memory flags to extend it.

Reply:
Also, I have not implemented yet the way to automate the indexing of these ones with docker & such. He you want, the subject's yours !

Reply:
For now I still manually call the file with node after giving the ES url in params

Reply:
I could do the deployement stuff, no pb, it's time to move away from python for dataset